autoload/clap/provider/grep.vim

```diff
@@ -29,11 +30,11 @@ function! s:draw_icon(line) abort
 endfunction
 
 function! s:cmd(query) abort
-  if !executable('rg')
-    call g:clap.abort('rg not found')
+  if !executable(s:grep_executable)
+    call g:clap.abort(s:grep_executable . ' not found')
     return
   endif
-  let cmd = 'rg -H --no-heading --vimgrep --smart-case '.s:grep_opts.' "'.a:query.'"'.(has('win32') ? ' .' : '')
+  let cmd = s:grep_executable.' '.s:grep_opts.' "'.a:query.'"'.(has('win32') ? ' .' : '')
```

Comment:
I'm not sure if `(has('win32') ? ' .' : '')` is also required for the other tools.

Reply:
Me neither, I dont have any knowledge about windows :(

Reply:
```diff
--- a/autoload/clap/provider/grep.vim
+++ b/autoload/clap/provider/grep.vim
@@ -8,6 +8,7 @@ let s:grep_delay = get(g:, 'clap_provider_grep_delay', 300)
 let s:grep_blink = get(g:, 'clap_provider_grep_blink', [2, 100])
 let s:grep_opts = get(g:, 'clap_provider_grep_opts', '-H --no-heading --vimgrep --smart-case')
 let s:grep_executable = get(g:, 'clap_provider_grep_executable', 'rg')
+let s:grep_cmd_format = get(g:, 'clap_provider_grep_cmd_format', '%s %s "%s"'.(has('win32') ? ' .' : ''))

 let s:old_query = ''
 let s:grep_timer = -1
@@ -34,7 +35,7 @@ function! s:cmd(query) abort
     call g:clap.abort(s:grep_executable . ' not found')
     return
   endif
-  let cmd = s:grep_executable.' '.s:grep_opts.' "'.a:query.'"'.(has('win32') ? ' .' : '')
+  let cmd = printf(s:grep_cmd_format, s:grep_executable, s:grep_opts, a:query)
   let g:clap.provider.cmd = cmd
   return cmd
 endfunction
```

Adding a new option `g:clap_provider_grep_cmd_format` should be a solution.

Do you mind applying this change in your change and add some doc for it?

Reply:
Hey, 

I have added the new option and updated the docs. Are the docs understandable?